client/landing/stepper/index.tsx

```diff
@@ -99,7 +99,7 @@ async function main() {
 	accessibleFocus();
 
 	const user = await initializeCurrentUser();
-	const userId = ( user as CurrentUser ).ID;
+	const userId = user ? ( user as CurrentUser ).ID : 0;
```

Comment:
Logic change seems reasonable, given that previously it would have "crashed" when `user` was `false`.

We don't need the typecasting now though.
```suggestion
	const userId = user ? user.ID : 0;
```

Reply:
In fact we should probably remove the type casing from lines 116 and 117 too. If this PR makes the `User` and `CurrentUser` types compatible, then we don't want any typecasting to hide the incompatibility.